UefiPayloadPkg/Library/PayloadEntryHobLib/Hob.c

```diff
@@ -109,6 +108,7 @@ CreateHob (
   VOID                        *Hob;
 
   HandOffHob = GetHobList ();
+  ASSERT (HandOffHob != NULL);
 
   //
   // Check Length to avoid data overflow.

```

Comment:
Another finding from Dhaval, the BuildResourceDescriptorHob () function in this file does not zero owner guid which could cause garbage data there.

please refer to below MdePkg one to zero owner guid:
https://github.com/tianocore/edk2/blob/master/MdePkg/Library/PeiHobLib/HobLib.c#L380